metadata-service/openapi-servlet/src/main/java/io/datahubproject/openapi/util/CondUpdateUtil.java

```diff
@@ -0,0 +1,33 @@
+package io.datahubproject.openapi.util;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+@Slf4j
+public class CondUpdateUtil {
+
+    // condUpdate String format: "urn1=createdOn1;urn2=createdOn2;...;urnN=createdOnN"
+
+    public static Map<String, Long> extractCondUpdate(String condUpdate) {
+        Map<String, Long> createdOnMap = new HashMap<>();
+        if (condUpdate != null) {
+            Arrays.stream(condUpdate.split(";"))
+                    .forEach(item -> {
+                        String[] values = item.split("=");
+                        if (values.length == 2) {
+                            try {
+                                createdOnMap.put(values[0], Long.parseLong(values[1]));
+                            } catch (Exception e) {
+                                log.warn("Invalid eTag: " + item);
```

Comment:
Not sure if eTag is the right name here any more? Maybe use `condition` if you want to be generic?